spec/models/user_spec.rb

```diff
@@ -54,4 +54,55 @@
       'Frank Poe'
     ]
   end
+
+  describe 'destroying' do
+    before :each do
+      @user1 = User.create(email:      'user1@example.com',
+                           password:   'passwd19873215',
+                           name:       'user1',
+                           last_name:  'user1',
+                           company:    'test company',
+                           department: 'test dep')
+      @user2 = User.create(email:      'user2@example.com',
+                           password:   'passdfsdswd19873215',
+                           name:       'user2',
+                           last_name:  'user2',
+                           company:    'test company',
+                           department: 'test dep')
+
+      @order_type = OrderType.create(code:   'test_order',
+                                     file:   "
+common: &common
+  label: ''
+  type: string
+
+order_type:
+  code: pizza_order
+  name: Pizza Order
+  fields:
+    customerName:
+      <<: *common
+                                                ",
+                                     fields: 'text',
+                                     name:   'qwerty')
+
+      @profile = Profile.create(user_id:       @user1.id,
+                                order_type_id: @order_type.id,
+                                data:          '{"qwe": 123}')
+
+      @order = Order.create(order_type_id: @order_type.id,
+                            user_id:       @user2.id,
+                            code:          'test_order')
+    end
+
+    it 'with profiles' do
+      @user1.destroy
+
+      expect { Profile.find(@profile.id) }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+
+    it 'restrictred if orders present' do
+      expect { @user2.destroy }.to raise_error(ActiveRecord::DeleteRestrictionError)
```

Comment:
you should also check if the user has been deleted or not

Reply:
Ok